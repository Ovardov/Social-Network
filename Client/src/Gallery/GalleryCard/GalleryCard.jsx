import React from 'react';
import styles from './gallery-card.module.scss';

function GalleryCard({ id, description, image, thumbnailUrl }) {
    return (
        <li className={styles.container}>
            <a className={styles['image-container']}>
                <img src={image} alt={description}/>
            </a>
        </li>
    )
}

export default GalleryCard;