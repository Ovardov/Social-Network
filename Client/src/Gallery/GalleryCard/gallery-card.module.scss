.container {
    list-style: none outside;
    overflow: hidden;
    position: relative;
    border-radius: 0.3em;
    box-shadow: 0 0 0.5em 0 rgba(0, 0, 0, 0.5);

    &:before {
        display: block;
        content: '';
        padding-top: 56.25%;
    }

    & .image-container {

        & img {
            display: block;
            width: 100%;
            height: auto;
            position: absolute;
            top: 50%;
            left: 50%;
            transform: translate(-50%, -50%);
        }
    }
}