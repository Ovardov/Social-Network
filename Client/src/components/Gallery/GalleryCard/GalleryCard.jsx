import React from 'react';
import styles from './gallery-card.module.scss';

function GalleryCard({ id, description, image, setIsOpen, index, setPhotoIndex }) {

    const handleClick = () => {
        setIsOpen(true);
        setPhotoIndex(index);
    }

    return (
        <li className={styles.container}>
            <button className={styles['image-container']} type="button" onClick={handleClick}>
                <img src={image} alt={description} />
            </button>
        </li>
    )
}

export default GalleryCard;