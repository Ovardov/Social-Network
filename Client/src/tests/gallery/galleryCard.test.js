import React from 'react'
import renderer from 'react-test-renderer';
import GalleryCard from '../../components/Gallery/GalleryCard/GalleryCard';

jest.mock("react-router-dom", () => ({
    Link: "Link"
}));

describe('GalleryCard component', () => {
    test('should render correcly with given data', () => {
        const component = renderer.create(
            <GalleryCard
                image="https://res.cloudinary.com/dxxq5xtsy/image/upload/v1576368427/cxfz26hn4gxeg36cv56q.jpg"
                describe="Aleksandar Ovardov"
            />
        );

        expect(component.toJSON()).toMatchSnapshot();
    });

});