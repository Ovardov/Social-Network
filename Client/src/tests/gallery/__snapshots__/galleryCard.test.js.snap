// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryCard component should render correcly with given data 1`] = `
<li
  className="container"
>
  <button
    className="image-container"
    onClick={[Function]}
    type="button"
  >
    <img
      src="https://res.cloudinary.com/dxxq5xtsy/image/upload/v1576368427/cxfz26hn4gxeg36cv56q.jpg"
    />
  </button>
</li>
`;
